traffic_ops/ort/atstccfg/remapdotconfig.go

```diff
@@ -0,0 +1,372 @@
+		return "", errors.New("getting delivery service regexes: " + err.Error())
+	}
+
+	dsRegexMap := map[tc.DeliveryServiceName][]tc.DeliveryServiceRegex{}
+	for _, dsRegex := range dsRegexes {
+		sort.Sort(DeliveryServiceRegexesSortByTypeThenSetNum(dsRegex.Regexes))
+		dsRegexMap[tc.DeliveryServiceName(dsRegex.DSName)] = dsRegex.Regexes
+	}
+
+	remapConfigDSData := []atscfg.RemapConfigDSData{}
+	for _, ds := range filteredDSes {
+		if ds.ID == nil || ds.Type == nil || ds.XMLID == nil || ds.DSCP == nil || ds.Active == nil {
+			continue // TODO log error?
+		}
+		// TODO sort by DS ID? the old Perl query does, but it shouldn't be necessary, except for determinism.
+		// TODO warn if no regexes?
+		for _, dsRegex := range dsRegexMap[tc.DeliveryServiceName(*ds.XMLID)] {
+			remapConfigDSData = append(remapConfigDSData, atscfg.RemapConfigDSData{
+				ID:                       *ds.ID,
+				Type:                     *ds.Type,
+				OriginFQDN:               ds.OrgServerFQDN,
+				MidHeaderRewrite:         ds.MidHeaderRewrite,
+				CacheURL:                 ds.CacheURL,
+				RangeRequestHandling:     ds.RangeRequestHandling,
+				RemapText:                ds.RemapText,
+				EdgeHeaderRewrite:        ds.EdgeHeaderRewrite,
+				SigningAlgorithm:         ds.SigningAlgorithm,
+				Name:                     *ds.XMLID,
+				QStringIgnore:            ds.QStringIgnore,
+				RegexRemap:               ds.RegexRemap,
+				FQPacingRate:             ds.FQPacingRate,
+				DSCP:                     *ds.DSCP,
+				RoutingName:              ds.RoutingName,
+				Pattern:                  util.StrPtr(dsRegex.Pattern),
+				RegexType:                util.StrPtr(dsRegex.Type),
+				Domain:                   util.StrPtr(serverCDNDomain), // note this is intentionally the CDN domain, not the DS or Server Domain. Must be the remap domain.
+				OriginShield:             ds.OriginShield,
+				ProfileID:                ds.ProfileID,
+				Protocol:                 ds.Protocol,
+				AnonymousBlockingEnabled: ds.AnonymousBlockingEnabled,
+				Active:                   *ds.Active,
+			})
+		}
+	}
+
+	serverProfileParams, err := GetProfileParameters(cfg, server.Profile)
+	if err != nil {
+		return "", errors.New("getting profile parameters from server (profile '" + server.Profile + ": " + err.Error())
+	}
+
+	serverPackageParamData := map[string]string{}
+	for _, param := range serverProfileParams {
+		if param.ConfigFile != "package" { // TODO put in const
+			continue
+		}
+
+		if param.Name == "location" { // TODO put in const
+			continue
+		}
+
+		paramName := param.Name
+		// some files have multiple lines with the same key... handle that with param id.
+		if _, ok := serverPackageParamData[param.Name]; ok {
+			paramName += "__" + strconv.Itoa(param.ID)
+		}
+		paramValue := param.Value
+		if paramValue == "STRING __HOSTNAME__" {
+			paramValue = server.HostName + "." + server.DomainName // TODO strings.Replace to replace all anywhere, instead of just an exact match?
+		}
```

Comment:
Are you sure this shouldn't be just checking for `__HOSTNAME__`? Does _all_ of `STRING __HOSTNAME__` get turned into the server's hostname? If that's true, it's undocumented.

Reply:
This is transliterating existing Perl: https://github.com/apache/trafficcontrol/blob/56ef95e1783360b4afb600b06cf138255fb34c89/traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm#L465

Reply:
It actually looks like this would never matter, since the only package it's interested in is something to do with dscp on which it does an exact string match. Bizarre, but as I understand it you're reluctant to make logic changes during translation.

Reply:
Yeah, we need to refactor all the config generation, they're all an overly-complicated mess, and I'm not sure half the logic is right.

Just hoping to do that in a separate PR, for safety.